### PR TITLE
handle rate limit config update event

### PR DESCRIPTION
**Pull request by mark4z**
**What this PR does**:
handle rate limit config update event

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -79,6 +79,20 @@ github.com/alibaba/sentinel-golang v1.0.2 h1:Acopq74hOtZN4MV1v811MQ6QcqPFLDSczTr
 github.com/alibaba/sentinel-golang v1.0.2/go.mod h1:QsB99f/z35D2AiMrAWwgWE85kDTkBUIkcmPrRt+61NI=
 github.com/aliyun/alibaba-cloud-sdk-go v1.61.18/go.mod h1:v8ESoHo4SyHmuB4b1tJqDHxfTGEciD+yhvOU/5s1Rfk=
 github.com/antihax/optional v1.0.0/go.mod h1:uupD/76wgC+ih3iEmQUL+0Ugr19nfwCT1kdvxnR2qWY=
+github.com/antihax/optional v1.0.0/go.mod h1:uupD/76wgC+ih3iEmQUL+0Ugr19nfwCT1kdvxnR2qWY=
```
Comment (AlexStocks) [inline go.sum:82]: pls run `go mod tidy` to compress the redundant lib

**Review by xiaoliu10 — APPROVED** [review]

**Review by ztelur — APPROVED** [review]

File: pkg/config/api_config.go [hunk 2]
```diff
@@ -136,11 +177,61 @@ func initAPIConfigFromKVList(kList, vList []string) error {
...
+	}
 
 	apiConfig = tmpApiConf
 	return nil
 }
 
+func initBaseInfoFromString(conf *fc.APIConfig, str string) error {
+	properties := make(map[string]string, 8)
+	if err := yaml.UnmarshalYML([]byte(str), properties); err != nil {
+		logger.Errorf("unmarshalYmlConfig error %v", err.Error())
```
Comment (AlexStocks) [inline pkg/config/api_config.go:196]: logger.Errorf("unmarshalYmlConfig error %s", err)

File: pkg/service/api/discovery_service.go [hunk 3]
```diff
@@ -269,3 +272,15 @@ func loadAPIFromMethods(fullPath string, methods []config.Method, headers map[st
 	}
 	return nil
 }
+
+func (ads *LocalMemoryAPIDiscoveryService) PluginPathChange(filePath string) {
```
Comment (AlexStocks) [inline pkg/service/api/discovery_service.go:276]: In Go, the receiver name should be as shorter as possible. Just one character should be enough.

**Review by xiaoliu10 — APPROVED** [review]

**Review by williamfeng323 — APPROVED** [review]

File: pkg/config/api_config.go [hunk 4]
```diff
@@ -122,32 +145,100 @@ func initAPIConfigFromKVList(kList, vList []string) error {
...
+		return err
+	}
 	if err := initAPIConfigServiceFromKvList(tmpApiConf, skList, svList); err != nil {
-		logger.Error("initAPIConfigServiceFromKvList error %v", err.Error())
+		logger.Errorf("initAPIConfigServiceFromKvList error %s", err.Error())
 		return err
 	}
 	if err := initAPIConfigMethodFromKvList(tmpApiConf, mkList, mvList); err != nil {
-		logger.Error("initAPIConfigMethodFromKvList error %v", err.Error())
+		logger.Errorf("initAPIConfigMethodFromKvList error %s", err.Error())
```
Comment (AlexStocks) [inline pkg/config/api_config.go:177]: pls delete the Error(), just the err is enough.

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**